README.md

```diff
@@ -6,3 +6,188 @@ Distributed service to interact with development boards.
+   ```
+   $ sudo apt install pdudaemon
+   ```
+
+1. We are going to use Static JWKS based authentication, as documented in the
+   [server README](boardswarm/README.md#static-jwks-based-authentication).
+   Install `rnbyc` so we can create the required tokens:
+   ```
+   $ sudo apt install rnbyc
+   ```
+
+1. Generate an EdDSA key:
+   ```
+   $ rnbyc -j -g  Ed25519  -x   -o private.jwks -p auth.jwks
+   ```
+   Ensure you keep the private key private and safe, it will be needed during
+   client configuration.
+
+1. Copy generated `auth.jwks` to `/etc/boardswarm/`
+   ```
+   $ sudo cp auth.jwks /etc/boardswarm/
+   ```
+
+For each development board you will need a section under `devices`. There is an
+example one provided in the example server configuration which can be used as a
+template.
+
+#### USB Serial Consoles
+
+1. Determine the USB serial port used as the console on the development board.
+
+1. Use the following commmand to determine the required `udev.ID_SERIAL` and
+   `udev.ID_USB_INTERFACE_NUM`:
+   ```
+   $ udevadm info <tty_device> | grep -i 'ID_SERIAL=\|ID_USB_INTERFACE_NUM='
+   ```
+
+#### DFU Booting
+
+1. Power on the development board in DFU mode and run:
+   ```
+   $ sudo dfu-util -l
+   dfu-util 0.11
+   
+   Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
+   Copyright 2010-2021 Tormod Volden and Stefan Schmidt
+   This program is Free Software and has ABSOLUTELY NO WARRANTY
+   Please report bugs to http://sourceforge.net/p/dfu-util/tickets/
+   
+   Found DFU: [0451:6165] ver=0200, devnum=45, cfg=1, intf=0, path="3-3.4", alt=1, name="SocId", serial="01.00.00.00"
+   Found DFU: [0451:6165] ver=0200, devnum=45, cfg=1, intf=0, path="3-3.4", alt=0, name="bootloader", serial="01.00.00.00"
+   ```
+
+1. Ensure that you have udev rules configuring the group ownership of usb
+   devices with the given USB vendor and product IDs (`0451:6165` in the above
+   example) to the boardswarm group.
+
+1. Use the following command to determine the required `ID_PATH`
+   ```
+   $ udevadm info /sys/bus/usb/devices/<path> | grep 'ID_PATH='
+   ```
+   Where `<path>` is the path found by `dfu-util` (`3-3.4` in the above
+   example).
+
+### Run the Server
+
+Reload systemd configi, start the boardswarm server and configure it to run at
+boot:
+```
+$ sudo systemctl daemon-reload
+$ sudo systectl enable boardswarm.service
+$ sudo systemctl start boardswarm.service
+```
+
+### Client Installation
+
+1. Copy the `boardswarm-cli` binary to `/usr/local/bin/` on the client
+   development machine.
+
+1. Create a user token (expiry 01-01-2025) using the previously generated
+   `private.jwks` and then copy it to the client machine:
+   ```
+   $ rnbyc  -s '{"exp": 1735686000 }' -K private.jwks > token.jwt
+   ```
+
+1. Initialise the connection to target server:
+   ```
+   $ boardswarm-cli -i <server_name> auth init --uri http://<server>:6683 --token-file token.jwt
+   ```
+
+### Client Usage
```

Comment:
would it not be better to point to a dedicated client document that's part of the boardswarm-cli documentation ? rather then document the client in the server codebase